src/main/java/frc/robot/subsystems/ClimberSubsystem.java

```diff
@@ -3,74 +3,78 @@
+  private final DoubleSolenoid m_rightSolenoid = new DoubleSolenoid(PneumaticsModuleType.REVPH,
+      ClimberConstants.rightForwardChannel, ClimberConstants.rightReverseChannel);
 
-    private final DoubleSolenoid m_leftSolenoid = new DoubleSolenoid(PneumaticsModuleType.REVPH,
-            ClimberConstants.leftForwardChannel, ClimberConstants.leftReverseChannel);
-    private final DoubleSolenoid m_rightSolenoid = new DoubleSolenoid(PneumaticsModuleType.REVPH,
-            ClimberConstants.rightForwardChannel, ClimberConstants.rightReverseChannel);
+  private final Compressor m_compressor = new Compressor(PneumaticsModuleType.REVPH);
 
-    private final Compressor m_compressor = new Compressor(PneumaticsModuleType.REVPH);
+  private boolean enableCompressor = true;
 
-    private boolean enableCompressor = true;
+  private Value m_state;
 
-    public ClimberSubsystem() {
-        m_leftSolenoid.set(kOff);
-        m_rightSolenoid.set(kOff);
+  public ClimberSubsystem() {
+        solenoidOff();
         m_compressor.disable();
         m_compressor.enableAnalog(ClimberConstants.minPressure, ClimberConstants.maxPressure);
     }
 
-    // Runs once every tick (~20ms)
-    public void periodic() {
-    }
+  // Runs once every tick (~20ms)
+  public void periodic() {
+    m_leftSolenoid.set(m_state);
+    m_rightSolenoid.set(m_state);
+  }
 
-    /**
-     * Sets the state of the solenoid to off
-     */
-    public void off() {
-        m_leftSolenoid.set(kOff);
-        m_rightSolenoid.set(kOff);
-    }
+  /**
+   * Sets the state of the solenoid to off
+   */
+  public void solenoidOff() {
+    m_state = kOff;
+  }
 
-    /**
-     * Extends both arms
-     */
-    public void forward() {
-        m_leftSolenoid.set(kForward);
-        m_rightSolenoid.set(kForward);
-    }
+  /**
+   * Extends both arms
+   */
+  public void forward() {
+    m_state = kForward;
+  }
 
-    /**
-     * Retracts both arms
-     */
-    public void reverse() {
-        m_leftSolenoid.set(kReverse);
-        m_rightSolenoid.set(kReverse);
-    }
-    /*
-     * Toggles the state of the climber
-     */
+  /**
+   * Retracts both arms
+   */
+  public void reverse() {
+    m_state = kReverse;
+  }
+  /*
+   * Toggles the state of the climber
+   */
 
-    public void toggle() {
-        m_leftSolenoid.toggle();
-        m_rightSolenoid.toggle();
+  public void toggle() {
+    if(m_state == kForward){
+      m_state = kReverse;
+    }else if(m_state == kReverse){
+      m_state = kForward;
     }
+  }
 
-    // Toggles the state of the compressor (on/off)
-    public void toggleCompresor() {
-        enableCompressor = !enableCompressor;
-        if (enableCompressor) {
-            m_compressor.enableAnalog(ClimberConstants.minPressure, ClimberConstants.maxPressure);
-        } else {
-            m_compressor.disable();
-        }
+  // Toggles the state of the compressor (on/off)
+  public void toggleCompresor() {
+    enableCompressor = !enableCompressor;
```

Comment:
Let's consistently use the m_ prefix for member variables. It might also be easier to understand if the name is compressorEnabled